src/nucleus.cpp

```diff
@@ -17,8 +23,20 @@ void Nucleus::activate(CellState *cellState) {
 	if (this->getSprite()->get_frame() == this->getSprite()->get_sprite_frames()->get_frame_count("activate") - 1)
 		this->getSprite()->stop();
 
-	if (cellState->getTotalNutrients() >= _reproductionNutrientThreshold) {
-		// UtilityFunctions::print("Nucleus activated: ", cellState->getTotalNutrients(), " >= ", _reproductionNutrientThreshold);
+	// Check that total resources are greater than or equal to the activation threshold and the reproduction cost
+	bool thresholdCondition = cellState->getTotalNutrients() >= _reproductionNutrientThreshold * cellState->getNutrientMaximum() && cellState->getTotalNutrients() >= cellState->getReproductionNutrientCost();
+	thresholdCondition = thresholdCondition && cellState->getTotalEnergy() >= _reproductionEnergyThreshold * cellState->getEnergyMaximum() && cellState->getTotalEnergy() >= cellState->getReproductionEnergyCost();
+
```

Comment:
Nitpick (feel free to ignore): Could instead define as nutrient & energy threshold conditions, then and them together in the if statement below which would be a bit clearer.